### PR TITLE
Ensure multiple Jakarta Data providers can coexist

### PR DESCRIPTION
**Pull request by KyleAure**
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

### CODE REVIEW

File: dev/io.openliberty.data.internal_fat_jpa_hibernate/test-applications/DataJPAHibernateIntegrationApp/src/test/jakarta/data/jpa/hibernate/integration/web/Segment.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    public int x1, y1;
+    public int x2, y2;
+
+    public static Segment of(int x1, int y1, int x2, int y2) {
+        Segment inst = new Segment();
+        inst.x1 = x1;
+        inst.y1 = y1;
+        inst.x1 = x2;
+        inst.y1 = y2;
```
Comment (njr-11) [inline dev/io.openliberty.data.internal_fat_jpa_hibernate/test-applications/DataJPAHibernateIntegrationApp/src/test/jakarta/data/jpa/hibernate/integration/web/Segment.java]:
```suggestion
        inst.x2 = x2;
        inst.y2 = y2;
```

File: dev/io.openliberty.data.internal_fat_jpa_hibernate/test-applications/DataJPAHibernateIntegrationApp/src/test/jakarta/data/jpa/hibernate/integration/web/Counties.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/*******************************************************************************
+ * Copyright (c) 2023,2024 IBM Corporation and others.
```
Comment (njr-11) [inline dev/io.openliberty.data.internal_fat_jpa_hibernate/test-applications/DataJPAHibernateIntegrationApp/src/test/jakarta/data/jpa/hibernate/integration/web/Counties.java]:
```suggestion
 * Copyright (c) 2024 IBM Corporation and others.
```

File: dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/Point_.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package test.jakarta.data.jpa.hibernate.integration.web;
+
+import jakarta.annotation.Generated;
+import jakarta.persistence.metamodel.EmbeddableType;
+import jakarta.persistence.metamodel.SingularAttribute;
+import jakarta.persistence.metamodel.StaticMetamodel;
+
+@StaticMetamodel(Point.class)
+@Generated("org.hibernate.processor.HibernateProcessor")
+public abstract class Point_ {
```
Comment (njr-11) [inline dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/Point_.java]: Can we delete this class?  It is probably left over from prior experimentation with entities.

File: dev/io.openliberty.data.internal_fat_jpa_hibernate/test-applications/DataJPAHibernateIntegrationApp/src/test/jakarta/data/jpa/hibernate/integration/web/County.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+/*******************************************************************************
+ * Copyright (c) 2023 IBM Corporation and others.
```
Comment (njr-11) [inline dev/io.openliberty.data.internal_fat_jpa_hibernate/test-applications/DataJPAHibernateIntegrationApp/src/test/jakarta/data/jpa/hibernate/integration/web/County.java]:
```suggestion
 * Copyright (c) 2024 IBM Corporation and others.
```

File: dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/Counties_.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
+package test.jakarta.data.jpa.hibernate.integration.web;
```
Comment (njr-11) [inline dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/Counties_.java:1]:
```suggestion
/*******************************************************************************
 * Copyright (c) 2024 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package test.jakarta.data.jpa.hibernate.integration.web;
```

File: dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/County_.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
+package test.jakarta.data.jpa.hibernate.integration.web;
```
Comment (njr-11) [inline dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/County_.java:1]:
```suggestion
/*******************************************************************************
 * Copyright (c) 2024 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package test.jakarta.data.jpa.hibernate.integration.web;
```

File: dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/Segment_.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
+package test.jakarta.data.jpa.hibernate.integration.web;
```
Comment (njr-11) [inline dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/Segment_.java:1]:
```suggestion
/*******************************************************************************
 * Copyright (c) 2024 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package test.jakarta.data.jpa.hibernate.integration.web;
```

File: dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/_County.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
+package test.jakarta.data.jpa.hibernate.integration.web;
```
Comment (njr-11) [inline dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/_County.java:1]:
```suggestion
/*******************************************************************************
 * Copyright (c) 2024 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package test.jakarta.data.jpa.hibernate.integration.web;
```

File: dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/_Segment.java [hunk 9]
```diff
@@ -0,0 +1,45 @@
+package test.jakarta.data.jpa.hibernate.integration.web;
```
Comment (njr-11) [inline dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/_Segment.java:1]:
```suggestion
/*******************************************************************************
 * Copyright (c) 2024 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License 2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     IBM Corporation - initial API and implementation
 *******************************************************************************/
package test.jakarta.data.jpa.hibernate.integration.web;
```

**Review by njr-11 — COMMENTED** [review]
Excellent tests. I added a few minor comments, mostly on copyrights. I think we still need to add those even when checking in generated code from Hibernate.

File: dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/Counties_.java (hunk 5 above)
Comment (KyleAure) [reply dev/io.openliberty.data.internal_fat_jpa_hibernate/generated-src/test/jakarta/data/jpa/hibernate/integration/web/Counties_.java:1]: Generated class cannot add copyright

**Comment by KyleAure** [comment]
#build

**Review by njr-11 — APPROVED** [review]

**Comment by KyleAure** [comment]
Pipeline: Failed with 2 known defects, and 1 timing issue
RTC Build: Failed with 3 known defects.

**Pull request merged**